Frontend/ui-common/src/Components/SelectListModal/Controller.purs

```diff
@@ -29,18 +32,22 @@ data Action = Button1 PrimaryButtonController.Action
             | ClearOptions
             
 type Config = {
-    cancelRideReasons :: Array CancellationReasons
+    selectionOptions :: Array OptionButtonList
     , primaryButtonTextConfig :: PrimaryButtonTextConfig
+    , headingTextConfig :: HeadingTextConfig
+    , subHeadingTextConfig :: SubHeadingTextConfig
+    , primaryButtonVisibility :: Boolean
+    , secondaryButtonVisibility :: Boolean
     , activeIndex :: Maybe Int
     , activeReasonCode :: Maybe String
-    , isCancelButtonActive :: Boolean
+    , isSelectButtonActive :: Boolean
     , isMandatoryTextHidden :: Boolean
     , isLimitExceeded :: Boolean
-    , headingText  :: String 
-    , subHeadingText :: String
+    , isGenderSelection :: Boolean
```

Comment:
@Akash2172 use some other name instead of isGenderSelection , as this has to be configurable and if there is any other scenario where we need this use case then there this name might not be serving the purpose.